distsql/distsql_test.go

```diff
@@ -20,142 +20,93 @@ import (
-	c.Assert(ok, IsTrue)
-	c.Assert(result.label, Equals, "dag")
-	c.Assert(result.sqlType, Equals, "general")
-	c.Assert(result.rowLen, Equals, len(colTypes))
-
-	resp, ok := result.resp.(*mockResponse)
-	c.Assert(ok, IsTrue)
-	resp.total = totalRows
-	resp.batch = batch
-
-	return result, colTypes
-}
-
-func (s *testSuite) TestSelectNormal(c *C) {
-	response, colTypes := s.createSelectNormal(1, 2, c, nil)
+func TestSelectNormal(t *testing.T) {
+	t.Parallel()
+	response, colTypes := createSelectNormal(t, 1, 2, nil, nil)
 
 	// Test Next.
 	chk := chunk.New(colTypes, 32, 32)
 	numAllRows := 0
 	for {
 		err := response.Next(context.TODO(), chk)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 		numAllRows += chk.NumRows()
 		if chk.NumRows() == 0 {
 			break
 		}
 	}
-	c.Assert(numAllRows, Equals, 2)
-	err := response.Close()
-	c.Assert(err, IsNil)
-	c.Assert(response.memTracker.BytesConsumed(), Equals, int64(0))
+	require.Equal(t, 2, numAllRows)
+	require.NoError(t, response.Close())
+	require.EqualValues(t, 0, response.memTracker.BytesConsumed())
 }
 
-func (s *testSuite) TestSelectMemTracker(c *C) {
-	response, colTypes := s.createSelectNormal(2, 6, c, nil)
+func TestSelectMemTracker(t *testing.T) {
+	t.Parallel()
+	response, colTypes := createSelectNormal(t, 2, 6, nil, nil)
 
 	// Test Next.
 	chk := chunk.New(colTypes, 3, 3)
 	err := response.Next(context.TODO(), chk)
-	c.Assert(err, IsNil)
-	c.Assert(chk.IsFull(), Equals, true)
-	err = response.Close()
-	c.Assert(err, IsNil)
-	c.Assert(response.memTracker.BytesConsumed(), Equals, int64(0))
+	require.NoError(t, err)
+	require.True(t, chk.IsFull())
+	require.NoError(t, response.Close())
+	require.EqualValues(t, 0, response.memTracker.BytesConsumed())
 }
 
-func (s *testSuite) TestSelectNormalChunkSize(c *C) {
-	s.sctx.GetSessionVars().EnableChunkRPC = false
-	response, colTypes := s.createSelectNormal(100, 1000000, c, nil)
-	s.testChunkSize(response, colTypes, c)
-	c.Assert(response.Close(), IsNil)
-	c.Assert(response.memTracker.BytesConsumed(), Equals, int64(0))
+func TestSelectNormalChunkSize(t *testing.T) {
+	t.Parallel()
+	sctx := newMockSessionContext()
+	sctx.GetSessionVars().EnableChunkRPC = false
+	response, colTypes := createSelectNormal(t, 100, 1000000, nil, sctx)
+	testChunkSize(t, response, colTypes)
+	require.NoError(t, response.Close())
+	require.EqualValues(t, 0, response.memTracker.BytesConsumed())
 }
 
-func (s *testSuite) TestSelectWithRuntimeStats(c *C) {
+func TestSelectWithRuntimeStats(t *testing.T) {
+	t.Parallel()
 	planIDs := []int{1, 2, 3}
-	response, colTypes := s.createSelectNormal(1, 2, c, planIDs)
-	if len(response.copPlanIDs) != len(planIDs) {
-		c.Fatal("invalid copPlanIDs")
-	}
-	for i := range planIDs {
-		if response.copPlanIDs[i] != planIDs[i] {
-			c.Fatal("invalid copPlanIDs")
-		}
-	}
+	response, colTypes := createSelectNormal(t, 1, 2, planIDs, nil)
+	require.ElementsMatch(t, planIDs, response.copPlanIDs, "invalid copPlanIDs")
```

Comment:
Maybe `Equal`,the original checks for the order, too.

Reply:
I think it is different from the above question, I used `ElementsMatch` to compare each object in each layer of the two collections, `ElementsMatch` requires lista, b must be array or slice.

Therefore, I think the semantics of this and the original check should be the same.

Reply:
I don't know if the original one is "ordered collections", but I tend to keep it. Because "ordered" is a more strict constraint. `ElementsMatch asserts that the specified listA(array, slice...) is equal to specified listB(array, slice...) ignoring the order of the elements.`. If you are sure that it is unordered, I can give LGTM.

Reply:
I understand what you mean, this is my problem. let me fix it.